Client/Frontend/Browser/RecentlyClosedTabs.swift

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public protocol RecentlyClosedTabs {
+    func addTab(url: NSURL, title: String?, faviconURL: String?)
+    func getTabs() -> [ClosedTab]
+    func clearTabs()
+    func count() -> Int
+}
+
+public class RecentlyClosedTabsStore: RecentlyClosedTabs {
+
+    public init() {
+    }
+
+    public func addTab(url: NSURL, title: String?, faviconURL: String?) {
+        if let recentlyClosedTab = ClosedTab(url: url, title: title ?? "", faviconURL: faviconURL ?? "") {
+            guard let recentlyClosedTabs = NSUserDefaults.standardUserDefaults().objectForKey("recentlyClosedTabs") else {
+                var recentlyClosedTabsArray = [ClosedTab]()
+                recentlyClosedTabsArray.append(recentlyClosedTab)
+                let archivedTabsArray = NSKeyedArchiver.archivedDataWithRootObject(recentlyClosedTabsArray)
+                NSUserDefaults.standardUserDefaults().setObject(archivedTabsArray, forKey: "recentlyClosedTabs")
+                return
+            }
```

Comment:
Note that the end of the `guard` is the same as the end of this method: we still want append the closed tab, serialize it, and save it to prefs. So instead of `guard`ing, just make an `if` statement that creates the array if it doesn't exist, and the code after that can be shared for both cases.